server/container/guice/memory-guice/src/main/java/org/apache/james/modules/mailbox/MemoryMailboxModule.java

```diff
@@ -78,6 +80,7 @@ protected void configure() {
         bind(MailboxManager.class).to(InMemoryMailboxManager.class);
         bind(MailboxACLResolver.class).to(UnionMailboxACLResolver.class);
         bind(GroupMembershipResolver.class).to(SimpleGroupMembershipResolver.class);
+        bind(new TypeLiteral<SystemMailboxesProvider<InMemoryId>>(){}).to(new TypeLiteral<SystemMailboxesProviderImpl<InMemoryId>>(){});
```

Comment:
I don't agree with you.

-> It's the only includes in here that are not in Mailbox

We had the same kind of problems with Methods during memory / cassandra project split.

Matthieu proposed this :

```
methods.addBinding().to(guiceGenericType.newGenericType(GetMailboxesMethod.class));
```

Seems nicer, in my opinion.